shell/platform/android/android_shell_holder.cc

```diff
@@ -0,0 +1,173 @@
+// Copyright 2018 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#define FML_USED_ON_EMBEDDER
+
+#include "flutter/shell/platform/android/android_shell_holder.h"
+
+#include <sys/resource.h>
+#include <sys/time.h>
+
+#include <sstream>
+#include <utility>
+
+#include "flutter/fml/message_loop.h"
+#include "flutter/shell/common/rasterizer.h"
+#include "flutter/shell/platform/android/platform_view_android.h"
+#include "lib/fxl/functional/make_copyable.h"
+
+namespace shell {
+
+static std::string CreateAndroidShellLabel(intptr_t handle) {
+  std::stringstream stream;
+  stream << "io.flutter.0x" << std::hex << handle;
```

Comment:
Remove the "io.flutter.0x" prefix - Linux thread names are limited to 16 characters

Reply:
Done. Still kept a count so that the number of the engine shows up in the native thread traces.

Reply:
Also updated the behavior for iOS.